File: javascript/src/challenges/average_pair.ts
// averagePair gets a sorted array of int and determine if there is a pair of values
// where the average of the pair equals the target average passed as second arg. O(n)
function averagePair(intArr: number[], targetAvg: number): boolean {
    // if the array has len zero returns false
    if (intArr.length === 0) return false

    // create a pointer to the end of the array
    let right = intArr.length - 1

    // loop thru the array
    for (let left = 0; left < intArr.length; left++) {
        // get the avg of the 2 pointers (left + right) / 2
        const avg = (intArr[left] + intArr[right]) / 2
        // if the avg is equal to the target return true
        if (avg === targetAvg) return true
        // if the right cross with left return false
        else if (left >= right) return false
        // if the avg is bigger than the decrease the right pointer
        else if (avg > targetAvg) right--
    }

    // if no avg found returns false
    return false
}

export default averagePair

File: python/challenges/average_pair.py
# average_pair gets a sorted list of integers and a target average,
# deterines if there is a pair of values in the list where the average of the pair
# equals the target average O(n)
def average_pair(int_list, target_avg):
    # if the len of int_list is zero returns false
    if len(int_list) == 0:
        return False

    # creates a pointer "right" to the end of the list
    right = len(int_list) - 1

    # loop thru the list
    for left in int_list:
        # calculates the avg (left value + right value / 2) and store in a var
        avg = (left + int_list[right]) / 2

        # if the avg matches with the target returns True
        if avg == target_avg:
            return True
        # else if the avg is bigger than the target, decrease the right pointer
        elif avg > target_avg:
            right -= 1

    # if the avg is not found returns false
    return False
